code/game/objects/structures/signs.dm

```diff
@@ -1111,3 +1116,31 @@
 
 /obj/structure/sign/bigname/seg_7
 	icon_state = "cyno_7"
+
+/obj/structure/sign/clock
+	name = "wall clock"
+	desc = "A basic wall clock, synced to the current system time."
+	icon_state = "clock"
+
+/obj/structure/sign/clock/Initialize()
+	. = ..()
+	desc += "\nThe clock shows that the time is [stationtime2text()]."

```

Comment:
As pointed out in discord, this appears to fetch the station-time _once_ at init and then never again, unless stationtime2text functions weirdly. Have you tested this several minutes after the object spawns?